Firmware/MotorControl/axis.cpp

```diff
@@ -397,29 +397,34 @@ bool Axis::run_homing() {
 
     error_ &= ~ERROR_MIN_ENDSTOP_PRESSED; // clear this error since we deliberately drove into the endstop
 
-    // pos_setpoint is the starting position for the trap_traj so we need to set it.
-    controller_.pos_setpoint_ = min_endstop_.config_.offset;
-    controller_.vel_setpoint_ = 0.0f;  // Change directions without decelerating
-
-    // Set our current position in encoder counts to make control more logical
-    encoder_.set_linear_count((int32_t)(controller_.pos_setpoint_ * encoder_.config_.cpr));
+    
+    if (encoder_.pos_estimate_.any() == std::nullopt || !encoder_.pos_estimate_.any().has_value()){
```

Comment:
There's a possible race condition here: `encoder_.pos_estimate_.any()` could become `std::nullopt` between this line and the next. To fix this you can read `encoder_.pos_estimate_.any()` into a local variable and then operate on that.

Reply:
Thanks for pointing that out. Modified it [here](https://github.com/odriverobotics/ODrive/pull/550/commits/7ac7c6ec58c6eabe7d832c7c9b105504c66c4242).

Reply:
Great, thanks! Looks good now.